### PR TITLE
feat: add 360 image list-getter, annotation default color and some adjustments

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:

Provides a few additions to the 360 API. Some of these are wanted for the next release of Data Explorer:

- Ability to get original color of an annotation
- Renamed `findImageAnnotation` -> `findImageAnnotations`
- Add `get360ImageCollections` method on viewer to list image collections
- Add new `remove360ImageCollection` method on viewer to remove
collection
- Deprecated `remove360Images` method on viewer
- Added some more TSDocs

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/360-images/src/annotation/Image360Annotation.ts [hunk 1]
```diff
@@ -19,6 +19,11 @@ export interface Image360Annotation {
    */
   getColor(): Color;
 
+  /**
+   * Get the randomly-generated color originally assigned to this annotation
+   */
+  getOriginalColor(): Color;
+
```
Comment (christjt) [inline viewer/packages/360-images/src/annotation/Image360Annotation.ts:26]: This is really confusing, what is the difference between get original color and get color?

File: viewer/packages/api/src/api-helpers/Image360ApiHelper.ts [hunk 2]
```diff
@@ -172,6 +180,20 @@ export class Image360ApiHelper {
...
+    if (
+      this._interactionState.currentImage360Entered !== undefined &&
+      collection.image360Entities.includes(this._interactionState.currentImage360Entered)
+    ) {
+      this.exit360Image();
+    }
+
+    pull(this._imageCollections, collection);
+
+    collection.image360Entities.forEach(entity => this._image360Facade.delete(entity as Image360Entity));
```
Comment (pramodcog) [inline viewer/packages/api/src/api-helpers/Image360ApiHelper.ts:193]: shouldn't it be `await Promise.all(collection.image360Entities.forEach(entity => this._image360Facade.delete(entity as Image360Entity)));` ?

File: viewer/packages/360-images/src/annotation/Image360Annotation.ts (hunk 1 above)
Comment (christjt) [reply viewer/packages/360-images/src/annotation/Image360Annotation.ts:26]: Conceptually I think it is kind of strange that we assign random color. We don't really do this for anything else. If anything it should at least be an optional thing, and if for some ungodly reason the user wants to override it and then recover it later I think that should be up to the app to cache that color. Would like your opinion on this though.

Comment (christjt) [reply viewer/packages/360-images/src/annotation/Image360Annotation.ts:26]: If not then at least call getOriginalColor -> getBaseColor, and (get/set)Color -> (get/set)OverrideColor

File: viewer/reveal.api.md [hunk 3]
```diff
@@ -382,7 +383,9 @@ export class Cognite3DViewer {
     on(event: 'sceneRendered', callback: SceneRenderedDelegate): void;
     get pointCloudBudget(): PointCloudBudget;
     set pointCloudBudget(budget: PointCloudBudget);
+    // @deprecated
     remove360Images(...image360Entities: Image360[]): Promise<void>;
+    remove360ImageSet(imageCollection: Image360Collection): void;
```
[the remark this answers is not in the document]
Comment (christjt) [reply viewer/reveal.api.md:388]: I don't think that is nessesary, can be easily accomplished from the app space with:
```ts
viewer.getImage360Collections().forEach(collection => viewer.remove360ImageSet(collection));
```

File: viewer/packages/360-images/src/annotation/Image360Annotation.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply viewer/packages/360-images/src/annotation/Image360Annotation.ts:26]: Alright, decided to remove this  function along with random coloring. From now on, all annotations will have the same color when loaded.

File: viewer/reveal.api.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (haakonflatval-cognite) [reply viewer/reveal.api.md:388]: Yes.. I think we're fine without it

**Review by christjt — APPROVED** [review]

**Pull request merged**